### PR TITLE
Update index.ts with OctokitOptions export

### LINKED ISSUE #645: [BUG]: OctokitOptions declared locally but not exported
### What happened?

I am trying to use OctokitOptions in a typescript project for DenoJS, but am getting the following error when using this import `import { OctokitOptions } from "@octokit/core";`

image

Not sure how to handle, thanks for the help.

### Versions

octokit/core: https://esm.sh/@octokit/core@v5.0.2

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mcgear**
Resolves #645 

---

### Before the change?

OctokitOptions is not currently exported for use in strong typing solutions.

-

### After the change?

OctokitOptions is exported from index.ts

-

### Pull request checklist

N/A

### Does this introduce a breaking change?

No breaking changes

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [x ] No

---

### CODE REVIEW

**Review by wolfy1339 — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -16,6 +16,8 @@ import type {
 } from "./types.js";
 import { VERSION } from "./version.js";
 
+export { OctokitOptions } from "./types.js";
```
Comment (wolfy1339) [inline src/index.ts:19]:
```suggestion
export type { OctokitOptions } from "./types.js";
```

**Comment by nickfloyd** [comment]
Hey @mcgear Let us know what you think about @wolfy1339's suggestion and once confirmed we can get this merged in and released. Thanks for the commits here ❤️ .

**Comment by mcgear** [comment]
Sorry about that, i put that in.

**Comment by wolfy1339** [comment]
Can you also merge the latest changes from the `main` branch?

**Comment by mcgear** [comment]
Done

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**